File: languages/typescript/Character.ts
import { BaseElement } from './base_elements';

export interface Character extends BaseElement {
  // Constitution
  physicality?: string | null;
  mentality?: string | null;
  height?: number | null;
  weight?: number | null;
  speciesIds?: string[] | null;
  traitsIds?: string[] | null;
  abilitiesIds?: string[] | null;
  // Origins
  background?: string | null;
  motivations?: string | null;
  birthDate?: number | null;
  birthplaceId?: string | null;
  languagesIds?: string[] | null;
  // World
  reputation?: string | null;
  locationId?: string | null;
  objectsIds?: string[] | null;
  institutionsIds?: string[] | null;
  // Personality
  charisma?: number | null;
  coercion?: number | null;
  competence?: number | null;
  compassion?: number | null;
  creativity?: number | null;
  courage?: number | null;
  // Social
  familyIds?: string[] | null;
  friendsIds?: string[] | null;
  rivalsIds?: string[] | null;
  // TTRPG
  level?: number | null;
  sTR?: number | null;
  dEX?: number | null;
  cON?: number | null;
  iNT?: number | null;
  wIS?: number | null;
  cHA?: number | null;
  hitPoints?: number | null;
}

File: languages/csharp_unity/Character.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Character : BaseElement
{
    // Constitution
    public string Physicality;
    public string Mentality;
    public int? Height;
    public int? Weight;
    public List<string> SpeciesIds; = new List<string>();
    public List<string> TraitsIds; = new List<string>();
    public List<string> AbilitiesIds; = new List<string>();
    // Origins
    public string Background;
    public string Motivations;
    public int? BirthDate;
    public string BirthplaceId;
    public List<string> LanguagesIds; = new List<string>();
    // World
    public string Reputation;
    public string LocationId;
    public List<string> ObjectsIds; = new List<string>();
    public List<string> InstitutionsIds; = new List<string>();
    // Personality
    public int? Charisma;
    public int? Coercion;
    public int? Competence;
    public int? Compassion;
    public int? Creativity;
    public int? Courage;
    // Social
    public List<string> FamilyIds; = new List<string>();
    public List<string> FriendsIds; = new List<string>();
    public List<string> RivalsIds; = new List<string>();
    // TTRPG
    public int? Level;
    public int? HitPoints;
    public int? STR;
    public int? DEX;
    public int? CON;
    public int? INT;
    public int? WIS;
    public int? CHA;
}
